### PR TITLE
Feat: global error handler to capture uncaught exceptions. Adds new event

### PR DESCRIPTION
**Pull request by cmgustavo**
https://github.com/a7ul/react-native-exception-handler 

Persist log for exceptions after the app crash.

Screenshot 2023-03-14 at 15 28 39

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,16 +1,42 @@
...
 import 'react-native-url-polyfill/auto'; // https://github.com/facebook/react-native/issues/23922#issuecomment-648096619
 import {enableFreeze} from 'react-native-screens';
+import { Analytics } from './src/store/analytics/analytics.effects';
+import { APP_VERSION } from './src/constants/config';
+import { GIT_COMMIT_HASH } from '@env';
 enableFreeze(true);
 
+const errorHandler = (e, isFatal) => {
+  if (isFatal) {
+    Analytics.track('BitPay App - Crashed App', { build: GIT_COMMIT_HASH, version: APP_VERSION});
```
Comment (cnguyen812) [inline index.js:20]: Should be: 
```
store.dispatch(Analytics.track(...));
```

and moved after the `store` declaration.

File: index.js [hunk 2]
```diff
@@ -1,16 +1,42 @@
...
+      [{
+        text: 'Close'
+      }]
+    );
+  }
+};
+setJSExceptionHandler(errorHandler, true);
+
+const exceptionHandler = () => {
+    Analytics.track('BitPay App - Crashed App', { build: GIT_COMMIT_HASH, version: APP_VERSION });
```
Comment (cnguyen812) [inline index.js:36]: Should be: 
```
store.dispatch(Analytics.track(...));
```

and moved after the `store` declaration.

File: package.json [hunk 3]
```diff
@@ -95,6 +95,7 @@
     "react-native-device-info": "8.4.7",
     "react-native-draggable-flatlist": "3.0.7",
     "react-native-error-boundary": "1.1.12",
+    "react-native-exception-handler": "2.10.10",
```
Comment (cnguyen812) [inline package.json:98]: Don't forget the Podfile.lock

File: index.js (hunk 1 above)
Comment (cnguyen812) [reply index.js:20]: Do we want to track the actual error as well? Or add to persistent logs?

**Review by cnguyen812 — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (cmgustavo) [reply index.js:23]: Yeah, good idea. I'll take a look. Thanks!

File: index.js (hunk 2 above)
Comment (cmgustavo) [reply index.js:39]: 👍🏼

File: index.js [hunk 4]
```diff
@@ -1,21 +1,66 @@
...
+const errorHandler = (e, isFatal) => {
+  if (isFatal) {
+    store.dispatch(
+      Analytics.track('BitPay App - Crashed App', {
+        build: GIT_COMMIT_HASH,
+        version: APP_VERSION,
+      }),
+    );
+    const errStr = e instanceof Error ? e.message : JSON.stringify(e);
+    LogActions.persistLog(LogActions.error(errStr));
```
Comment (cnguyen812) [inline index.js:36]: This needs to be dispatched as an store action

```
store.dispatch(LogActions...)
```

**Review by cnguyen812 — DISMISSED** [review]

File: index.js (hunk 4 above)
Comment (cmgustavo) [reply index.js:36]: Yes, fixed. Thanks!

File: index.js [hunk 5]
```diff
@@ -2,22 +2,67 @@ import 'react-native-get-random-values'; // must import before @ethersproject/sh
 import '@ethersproject/shims';
 import './shim';
 import '@walletconnect/react-native-compat';
-import {AppRegistry} from 'react-native';
+import {AppRegistry, Alert} from 'react-native';
+import 'react-native-get-random-values';
```
Comment (msalcala11) [inline index.js:6]: Why is this import needed?

Comment (cmgustavo) [reply index.js:6]: Oh... it's duplicated (from line 1). Good catch!

**Review by msalcala11 — APPROVED** [review]
Works great!
